tests/test_search.py

```diff
@@ -665,7 +668,8 @@ def test_detection(self, shortname, text):
         param(text="DECEMBER 21 19.87 87",
               languages=None,
               settings=None,
-              expected=[('DECEMBER 21 19', datetime.datetime(2019, 12, 21, 0, 0))]
+              expected=[('DECEMBER', datetime.datetime(2019, 12, 21, 0, 0)),
+                        ('19.87', datetime.datetime(1987, 12, 19, 0, 0))]
```

Comment:
Changing this test does not look right. Interpreting `19.87` as the year `1987` definitely looks wrong.

Reply:
This is a bit odd because the parser can't figure out anything from the full string `DECEMBER 21 19.87 87`. So it splits it up into a bunch of different substrings and tries to figure out what returns the best results. In this case, if it splits on spaces it ends up with the most results.

When trying to parse `19.87`, it splits on the `.`, tries to figure out what `19` could represent, comes up with a day of the month, then tries to figure out what `87` could mean, and comes up with a year. It then guesses that the year `87` refers to `1987` rather than `2087 and returns that.

You can replicate this by just running `search_dates("19.87")`. Or if you run `search_dates("21.87")` it returns the 21st of the month, still in 1987. I can edit the test to use 21 instead of 19 to make it a bit more clear if you want.